lib/beaker-pe/install/pe_utils.rb

```diff
@@ -535,6 +505,82 @@ def do_install hosts, opts = {}
+        #
+        # @note that these steps aren't necessary for all hosts. Specifically,
+        #   'agent_only' hosts do not require these steps to be executed.
+        #
+        # @param [Array<Host>] hosts Hosts to prepare
+        # @param [Hash{Symbol=>String}] local_options Local options, used to
+        #   pass misc configuration required for the prep steps
+        #
+        # @return nil
+        def prepare_hosts(hosts, local_options={})
+          use_all_tar = ENV['PE_USE_ALL_TAR'] == 'true'
+          hosts.each do |host|
+            host['pe_installer'] ||= 'puppet-enterprise-installer'
+            if host['platform'] !~ /windows|osx/
+              platform = use_all_tar ? 'all' : host['platform']
+              version = host['pe_ver'] || local_options[:pe_ver]
+              host['dist'] = "puppet-enterprise-#{version}-#{platform}"
+            elsif host['platform'] =~ /osx/
+              version = host['pe_ver'] || local_options[:pe_ver]
+              host['dist'] = "puppet-enterprise-#{version}-#{host['platform']}"
+            elsif host['platform'] =~ /windows/
+              version = host[:pe_ver] || local_options['pe_ver_win']
+              is_config_32 = true == (host['ruby_arch'] == 'x86') || host['install_32'] || local_options['install_32']
+              should_install_64bit = !(version_is_less(version, '3.4')) && host.is_x86_64? && !is_config_32
+              #only install 64bit builds if
+              # - we are on pe version 3.4+
+              # - we do not have install_32 set on host
+              # - we do not have install_32 set globally
+              if !(version_is_less(version, '3.99'))
+                if should_install_64bit
+                  host['dist'] = "puppet-agent-#{version}-x64"
+                else
+                  host['dist'] = "puppet-agent-#{version}-x86"
+                end
+              elsif should_install_64bit
+                host['dist'] = "puppet-enterprise-#{version}-x64"
+              else
+                host['dist'] = "puppet-enterprise-#{version}"
+              end
+            end
+            host['working_dir'] = host.tmpdir(Time.new.strftime("%Y-%m-%d_%H.%M.%S"))
+          end
+        end
+
+        # Gets the puppet-agent version, hopefully from the host or local options.
+        # Will fall back to reading the `aio_agent_build` property on the master
+        # if neither of those two options are passed
+        #
+        # @note This method does have a side-effect: if it reads the
+        #   `aio_agent_build` property from master, it will store it in the local
+        #   options hash so that it won't have to do this more than once.
+        #
+        # @param [Beaker::Host] host Host to get puppet-agent for
+        # @param [Hash{Symbol=>String}] local_options local method options hash
+        #
+        # @return [String] puppet-agent version to install
+        def get_puppet_agent_version(host, local_options={})
```

Comment:
A few comments here, not all that need to be addressed in this PR:

1) We continue to call things `puppet_*` when we really mean AIO. We should really figure out a way to clean that so that we can say what we mean. </end rant>

2) This method uses the `aio_agent_build` key that puppet facts returns; from what I can tell, we actually want to use `aio_agent_version` that is identical to `aio_agent_build` except for it doesn't contain the git sha for the version number.
Example build: "aio_agent_build": "1.5.3.245.g9bd94ac"
Example version: "aio_agent_version": "1.5.3.245"

My understanding is that we could use the AIO version to build a path to the tarballs to download; I'm unclear on how we would use the build number in this way.